plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/connection/DataSourceVariableResolver.java

```diff
@@ -47,6 +49,10 @@ protected DBPConnectionConfiguration getConfiguration() {
 
     @Override
     protected boolean isResolveSystemVariables() {
+        DBPApplication application = DBWorkbench.getPlatform().getApplication();
```

Comment:
I think we add function isEnvironmentVariablesAccessible into DBPApplication to make it more general. 
Usage of `instanceof DBPApplicationDesktop` is a silly hack.